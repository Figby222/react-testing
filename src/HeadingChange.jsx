import { useState } from 'react';

function HeadingChange() {
    const [ heading, setHeading ] = useState('I like cookies');

    const clickHandler = () => {
        setHeading('I like pizza');
    }

    return (
        <>
            <button type="button" onClick={clickHandler}>Change heading text</button>
            <h1>{heading}</h1>
        </>
    )
}

export default HeadingChange;