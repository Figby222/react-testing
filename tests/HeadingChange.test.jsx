import { render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import HeadingChange from "../src/HeadingChange.jsx";

describe("HeaderChange component", () => {
    it('renders I like cookies', () => {
        const { container } = render(<HeadingChange />);
        expect(container).toMatchSnapshot();
    });

    it('renders I like pizza after button click', async () => {
        const user = userEvent.setup();

        render(<HeadingChange />);
        const button = screen.getByRole("button", { name: "Change heading text" });

        await user.click(button);

        expect(screen.getByRole("heading").textContent).toMatch(/i like pizza/i);
    });
});