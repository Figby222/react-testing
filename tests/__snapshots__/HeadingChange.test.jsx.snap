// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderChange component > renders I like cookies 1`] = `
<div>
  <button
    type="button"
  >
    Change heading text
  </button>
  <h1>
    I like cookies
  </h1>
</div>
`;
